File: server/src/CardService.ts
import { Card } from './store';
import { DeckShoe } from './DeskShoe';

const NUM_DECKS_IN_SHOE = 7;
export class CardService {
  deckShoe: DeckShoe;

  constructor() {
    this.deckShoe = new DeckShoe(NUM_DECKS_IN_SHOE);
  }

  nextCard(): Card {
    return this.deckShoe.getNextCard();
  }

  shuffle() {
    this.deckShoe.shuffle();
  }
}

export const theCardService = new CardService();

File: django/blackjack/store/card_service.py
from blackjack.state.card import Card
from blackjack.store.deck_shoe import DeckShoe

NUM_DECKS_IN_SHOE = 7
class CardService:
    def __init__(self) -> None:
        self.deckShoe = DeckShoe(NUM_DECKS_IN_SHOE)

    def nextCard(self) -> Card:
        return self.deckShoe.getNextCard()
    
    def shuffle(self):
        self.deckShoe.shuffle()
        